### PR TITLE
Feature/ecdsa

### PR DESCRIPTION
**Pull request by cyberzac**
Added Elliptic Curve

### CODE REVIEW

File: crypto-bouncycastle/src/main/scala/cryptic/crypto/ECIES.scala [hunk 1]
```diff
@@ -0,0 +1,32 @@
+package cryptic.crypto
+
+import java.security._
+
+import cryptic.{CipherText, Decrypt, Encrypt, PlainText}
+import javax.crypto.Cipher
+import org.bouncycastle.jce.provider.BouncyCastleProvider
+
+/**
+  * RSA encryption depends  private and public keys.
```
Comment (odd) [inline crypto-bouncycastle/src/main/scala/cryptic/crypto/ECIES.scala]: Too much interspace makes us feel lonely.

File: crypto-bouncycastle/src/main/scala/cryptic/crypto/ECIES.scala [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  * The public key should be implicitly available for encryption and
+  * the private key for decryption
+  */
+object ECIES {
+  Security.addProvider(new BouncyCastleProvider())
```
Comment (odd) [inline crypto-bouncycastle/src/main/scala/cryptic/crypto/ECIES.scala]: Is BouncyCastle the chosen one or should it be possible to use other providers?

**Comment by odd** [comment]
Please use all lowercase branch names from here on 🤓

File: crypto-bouncycastle/src/main/scala/cryptic/crypto/ECIES.scala (hunk 1 above)
Comment (cyberzac) [reply crypto-bouncycastle/src/main/scala/cryptic/crypto/ECIES.scala]: Will remove, and fix the copy waste comment...

File: crypto-bouncycastle/src/main/scala/cryptic/crypto/ECIES.scala (hunk 2 above)
Comment (cyberzac) [reply crypto-bouncycastle/src/main/scala/cryptic/crypto/ECIES.scala]: I think that Bounce Castle is the only one available. And it is simple to use your own.

File: crypto-bouncycastle/src/main/scala/cryptic/crypto/ECIES.scala (hunk 1 above)
Comment (cyberzac) [reply crypto-bouncycastle/src/main/scala/cryptic/crypto/ECIES.scala]: Can we have rules for Scalafmt checked in?

Comment (odd) [reply crypto-bouncycastle/src/main/scala/cryptic/crypto/ECIES.scala]: Sure, that's a great idea.

**Pull request merged**